File: ios/NorthAx/Models/SportTarget.swift
import Foundation

enum GoalType: String, Codable {
    case raceTime          // Running: distance + finish time
    case powerHold         // Cycling: hold a power zone for a duration
    case distanceAvgSpeed  // Cycling: distance at an average speed
}

/// One structured goal per sport, fed to the AI planner and the post-sync
/// progress analysis. Flat struct + `goalType` discriminator, matching the
/// AthleteThresholds style.
struct SportTarget: Codable, Equatable {
    var goalType: GoalType
    var targetDate: Date
    var distanceKm: Double?      // raceTime, distanceAvgSpeed
    var finishTimeSec: Int?      // raceTime
    var zone: Int?               // powerHold (1-5)
    var holdMinutes: Int?        // powerHold
    var avgSpeedKmh: Double?     // distanceAvgSpeed
}

/// Latest AI goal-progress verdict for one targeted sport.
struct GoalCheck: Identifiable, Equatable {
    enum Verdict: String {
        case onTrack = "on_track"
        case behind
        case ahead
    }

    var domain: TrainingDomain
    var verdict: Verdict
    var summary: String
    var recommendReplan: Bool
    var analyzedAt: Date

    var id: TrainingDomain { domain }
}

File: android/app/src/main/java/app/northax/domain/model/SportTarget.kt
package app.northax.domain.model

import java.time.Instant
import java.time.LocalDate

enum class GoalType(val raw: String) {
    RaceTime("raceTime"),                  // Running: distance + finish time
    PowerHold("powerHold"),                // Cycling: hold a power zone for a duration
    DistanceAvgSpeed("distanceAvgSpeed");  // Cycling: distance at an average speed

    companion object {
        fun fromRaw(raw: String): GoalType? = entries.firstOrNull { it.raw == raw }
    }
}

/**
 * One structured goal per sport, fed to the AI planner and the post-sync
 * progress analysis. Flat class + `goalType` discriminator.
 */
data class SportTarget(
    val goalType: GoalType,
    val targetDate: LocalDate,
    val distanceKm: Double? = null,   // raceTime, distanceAvgSpeed
    val finishTimeSec: Int? = null,   // raceTime
    val zone: Int? = null,            // powerHold (1-5)
    val holdMinutes: Int? = null,     // powerHold
    val avgSpeedKmh: Double? = null,  // distanceAvgSpeed
)

/** Latest AI goal-progress verdict for one targeted sport. */
data class GoalCheck(
    val domain: TrainingDomain,
    val verdict: Verdict,
    val summary: String,
    val recommendReplan: Boolean,
    val analyzedAt: Instant,
) {
    enum class Verdict(val raw: String) {
        OnTrack("on_track"), Behind("behind"), Ahead("ahead");

        companion object {
            fun fromRaw(raw: String): Verdict? = entries.firstOrNull { it.raw == raw }
        }
    }
}
